src/Symfony/Bridge/Twig/UndefinedCallableHandler.php

```diff
@@ -64,34 +67,34 @@ class UndefinedCallableHandler
         'workflow' => 'enable "framework.workflows"',
     ];
 
-    public static function onUndefinedFilter(string $name): bool
+    public static function onUndefinedFilter(string $name)
     {
         if (!isset(self::FILTER_COMPONENTS[$name])) {
             return false;
         }
 
-        self::onUndefined($name, 'filter', self::FILTER_COMPONENTS[$name]);
-
-        return true;
+        throw new SyntaxError(self::onUndefined($name, 'filter', self::FILTER_COMPONENTS[$name]));
```

Comment:
What’s the significance of changing from a return to a throw? That part looks unrelated (but I’m sure it is related)

Reply:
The return type should be `TwigFilter|false`, `bool` is just wrong

Reply:
now clarified with a docblock

Reply:
Well, `return true` was actually dead code anyway, as the old `onUndefined` was never returning (but was not annotated as `@return never` so static analysis would not know it)